### PR TITLE
Update product image selection page

### PR DESCRIPTION
**Pull request by fhennig**
- give a simple "TLDR" style config snippet right in the beginning
- fix some typos and wordings
- explicitly state where to find base images for custom images

### CODE REVIEW

File: modules/concepts/pages/product-image-selection.adoc [hunk 1]
```diff
@@ -1,20 +1,35 @@
...
+
+[source,yaml]
+----
+spec:
+  image:
+    productVersion: 1.2.3 <.>
+    # stackableVersion: 24.3.0 optional <.>
+----
+<.> The version of your product - consult the product operator documentation to find out about supported product versions.
+<.> The version of the Stackable Data Platform - simply omit this to use the operator version.
```
Comment (razvan) [inline modules/concepts/pages/product-image-selection.adoc:17]:
```suggestion
<.> The version of the Stackable Data Platform. If you omit it, the operator will use it's own version together with the product version to select the product image.
```

File: modules/concepts/pages/product-image-selection.adoc [hunk 2]
```diff
@@ -1,20 +1,35 @@
...
+
+This page explains the different ways of specifying product images and the components that are involved,
+starting from the simple way of using the default images to explaining how to use custom or mirrored registries as well as custom images.
+
+== Product and Stackable version explained
+
+All products of the Stackable Data Platform run on Kubernetes and are managed by Stackable Operators and one operator is responsible for exactly one product like Apache Spark.
```
Comment (razvan) [inline modules/concepts/pages/product-image-selection.adoc:24]:
```suggestion
All products of the Stackable Data Platform run on Kubernetes and are managed by Stackable Operators. One operator is responsible for exactly one product, for example Apache Spark, but can manage multiple instances (with different versions) of the product at the same time.
```

File: modules/concepts/pages/product-image-selection.adoc [hunk 3]
```diff
@@ -1,20 +1,35 @@
...
+The operators read the Stacklet definition and deploy appropriate StatefulSets which lead to the creation of Pods, which in the end require _container images_ to run.
```
Comment (razvan) [inline modules/concepts/pages/product-image-selection.adoc:25]:
```suggestion
The operators transform a Stacklet definition (for example a SparkApplication object) into Kubernetes native objects. Some of these objects are Pods and they  require _container images_ to run.
```

File: modules/concepts/pages/product-image-selection.adoc [hunk 4]
```diff
@@ -60,29 +75,33 @@ At the bottom of this page in the <<_common_scenarios, common scenarios>> sectio
...
+    # stackableVersion: 23.7.0 # optional <.>
+----
+<.> The version of your product - consult the product operator documentation to find out about supported product versions.
+<.> The version of the Stackable Data Platform - simply omit this to use the operator version.
+
+NOTE: If the Kubernetes cluster does not have internet access, a xref:_custom_docker_registry[] or xref:_custom_images[] can be used.
 
-Currently, you need to specify the product version. This can be found on the xref:operators:supported_versions.adoc[list of supported product versions] or on the website of the product itself.
-This requirement might be relaxed in the future, as every platform release will ship wth a recommended product versions, which will be used by default.
+You only need to specify the product version but _can_ also specify an exiplicit Stackable version.
```
Comment (razvan) [inline modules/concepts/pages/product-image-selection.adoc:92]:
```suggestion
You only need to specify the product version but _can_ also specify an explicit Stackable version.
```

File: modules/concepts/pages/product-image-selection.adoc [hunk 5]
```diff
@@ -60,29 +75,33 @@ At the bottom of this page in the <<_common_scenarios, common scenarios>> sectio
...
+the product version can be found on the xref:operators:supported_versions.adoc[list of supported product versions] or on the product operator documentation page.
+The requirement to specify a product version might be relaxed in the future, as every platform release might ship with a recommended product version, which might be used by default.
```
Comment (razvan) [inline modules/concepts/pages/product-image-selection.adoc:94]: I would delete this statement. It's a promise and it's not actionable for users.

File: modules/concepts/pages/product-image-selection.adoc [hunk 6]
```diff
@@ -115,26 +134,25 @@ spec:
     productVersion: 3.3.1
 ----
 
-The Stackable Operators configure their respective products based on the product version.
-This affects e.g. configuration properties or available features.
-Therefore, the operators are dependent on the product and its product version contained in the custom image.
-It's your responsibility to put in the correct product version.
+Even though the product version is not used anymore for image selection, you still need to provide it, as the Stackable Operators configure their respective products based on the product version.
```
Comment (razvan) [inline modules/concepts/pages/product-image-selection.adoc:137]:
```suggestion
Even though the product version is not used anymore for image selection, you still need to provide it, as the operators configure their respective products based on the product version.
```

File: modules/concepts/pages/product-image-selection.adoc (hunk 5 above)
Comment (fhennig) [reply modules/concepts/pages/product-image-selection.adoc:94]: I agree, will do

Comment (fhennig) [reply modules/concepts/pages/product-image-selection.adoc]: https://github.com/stackabletech/documentation/pull/567/commits/618e2809799ca1ca9719fdc56584366f168c9230

**Review by razvan — APPROVED** [review]

**Pull request merged**